assets/app/views/directives/osc-key-values.html

```diff
@@ -2,18 +2,27 @@
     <div class="form-inline labels-edit" ng-show="editable">
       <form class="edit-label" name="form" novalidate>
         <div class='form-group' ng-class="{'has-error': form.key.$error.oscKeyValid}">
-          <input class="form-control" type="text" name="key" placeholder="{{keyTitle}}" ng-model="key" 
+          <input class="form-control"
+                 type="text"
+                 name="key"
+                 placeholder="{{keyTitle}}"
+                 ng-model="key"
+                 ng-keyup="toggleAdd()"
                  osc-input-validator="key">
         </div>
         <div class="form-group" ng-class="{'has-error': form.value.$error.oscValueValid}">
```

Comment:
It looks like both create.html & createProject.html have a similar mechanism, but both are using built-in directives rather than a custom validation function.  If I refactored this toggle mechanism, I probably should do a couple things:

- Use min-length on the key & value
- Create a custom validation directive to detect if the key is already in the read-only and entries lists & invalidate the model.  In this context it looks like it would have to support a string list, array list, hash.

I'm not sure if there is a good, clean way to do this as a directive instead of a controller function.